#卒業論文
-----
###第5章「シミュレーション分析」

In [1]:
# coding: UTF-8
%matplotlib inline
import itertools
import numpy as np
import matchfuncs as mf

In [2]:
prop_num = 345
resp_num = 38
alpha = 0.5
beta = 1
prop_caps = [1 for i in range(230)] + [2 for i in range(115)]
resp_caps = [12 for i in range(25)] + [6 for i in range(13)]
list_length = 2

In [3]:
bos_data = np.zeros(7)
daadd_data = np.zeros(7)
da_data = np.zeros(7)

In [4]:
t = 10000
for i in range(t):
    prop_prefs, resp_prefs, popularity, grade = mf.MakeCVprefs(prop_num, resp_num, alpha, beta)
    bos_data += np.asarray(mf.Comp('BOS', prop_prefs, prop_prefs, resp_prefs, resp_caps, prop_caps, list_length))
    daadd_data += np.asarray(mf.Comp('DAAdd', prop_prefs, prop_prefs, resp_prefs, resp_caps, prop_caps, list_length))
    da_data += np.asarray(mf.Comp('DA', prop_prefs, prop_prefs, resp_prefs, resp_caps, prop_caps))
bos_data /= t
daadd_data /= t
da_data /= t

In [5]:
print(bos_data)
print(daadd_data)
print(da_data)

[ 240.3507     345.           4.2881     121.3594517  116.8141     106.4156
  825.7327   ]
[ 216.0581     345.           2.7225     110.3444166  152.6249     153.0434
  934.6192   ]
[  0.00000000e+00   3.45000000e+02   9.24105000e+00   1.27819164e+02
   6.08445000e+01   0.00000000e+00   1.31100000e+04]


|| 制度1 | 制度2 | 制度3 |
|:-----:|:-----------:|:------------:|:------------:|
|**安定性**| 240.35| 216.06 | 0 |
|**耐戦略性**| 345 | 345 | 345 |
|**効率性(生徒)**| 4.29 |  2.72 |  9.24  |
|**効率性(ゼミ)**| 121.36 | 110.34 | 127.81 |
|**衡平性(生徒)**| 116.81 | 152.62 | 60.84 |
|**衡平性(ゼミ)**| 106.42 | 153.04 | 0 |
|**実現可能性**| 825.73 | 934.62 | 13110 |